### PR TITLE
[Torch Dialect] add folder for aten.any.bool

### PR DESCRIPTION
**Pull request by JianzheXiao**
Aviod error occured in runtime.assert in stablehlo

### CODE REVIEW

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -1437,6 +1437,23 @@ OpFoldResult AtenBoolIntOp::fold(FoldAdaptor adaptor) {
...
 }
 
+//===----------------------------------------------------------------------===//
+// AtenAnyBoolOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenAnyBoolOp::fold(FoldAdaptor adaptor) {
+  SmallVector<bool> inputList;
+  if (!matchPattern(getSelf(), m_TorchListOfConstantBools(inputList)))
+    return nullptr;
```
Comment (qingyunqu) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1447]: I think we can implement a more aggressive folder in future. We can match a `torch.List.Construct` and if we find that any operand of `torch.List.Construct` is `torch.constant.bool true`, just fold it to `true`.

Comment (JianzheXiao) [reply lib/Dialect/Torch/IR/TorchOps.cpp]: Fixed

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 2]
```diff
@@ -1437,6 +1437,22 @@ OpFoldResult AtenBoolIntOp::fold(FoldAdaptor adaptor) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// AtenAnyBoolOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenAnyBoolOp::fold(FoldAdaptor adaptor) {
+  auto inputConsturct = getSelf().getDefiningOp<Torch::PrimListConstructOp>();
```
Comment (ramiro050) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1445]: Drive by comment: need to check that the defining op is indeed a `PrimListConstructOp` and that the list is not mutated. Similar to this: 

https://github.com/llvm/torch-mlir/blob/139e820e15fc441fe90f595ee927035304b5b606/lib/Dialect/Torch/IR/TorchOps.cpp#L1931-L1933

Comment (ramiro050) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1445]: nit: inputConst**r**u~r~ct

**Comment by JianzheXiao** [comment]
@powderluv Hi! Could you please help fix the CI failure (git checkout error)?

**Pull request merged**